source/PlayerInfo.h

```diff
@@ -189,8 +190,16 @@ class PlayerInfo {
 	
 	// Access the "condition" flags for this player.
 	int64_t GetCondition(const std::string &name) const;
-	std::map<std::string, int64_t> &Conditions();
-	const std::map<std::string, int64_t> &Conditions() const;
+	// Set a "condition" flag to the given value. Returns true on success,
+	// false on failure.
+	bool SetCondition(const std::string &name, int64_t value);
+	bool AddCondition(const std::string &name, int64_t value);
+	bool EraseCondition(const std::string &name);
+	void EraseManualByPrefix(const std::string &prefix);
+	// Direct access to "condition" flags data.
+	ConditionsStore &Conditions();
+	const ConditionsStore &Conditions() const;
+	const std::map<std::string, int64_t> &GetManualConditions() const;
```

Comment:
Just by reading the names, I'm not exactly sure what "manual" is suppose to mean in these method names. Would "raw" be a better term to use? "GetRawConditions" and "EraseRawByPrefix"? Or maybe "direct" for "GetDirectConditions" etc.

Reply:
This is the set of conditions that is not On-Demand. I guess `GetNotOnDemandConditions` would also be an appropriate name for this function, but it sounds a bit odd as well.
`Raw` doesn't sound any more clear than `Manual` to me. `Direct` however also okay for me.

Reply:
We could also consider if `On Demand` is the best name for conditions that are based on other data, maybe we should rename those to `Derived Conditions`, in which case we can use `GetNonDerivedConditions` as name here (and add a comment that non-derived conditions are stored in a savegame)

Reply:
Okay, I see where the term comes from. Some conditions are “automatic” (maintained by the source code and writing to them just results in that being overwritten), meaning those conditions which aren’t automatic are manual. 

Are you expecting to have GetAutomaticConditions or similar? If not then maybe it’d be better to just have this be GetConditionsSubset. That way the method name makes it clear that these aren’t all the conditions (or rather, won’t be all the conditions), and then a comment can explain exactly what that subset consists of.

Reply:
Let's go for `Derived Conditions` for conditions that are based on other parts of the code and let's use `Primary Conditions` for conditions that are set directly not based on any data in any other class. I'll update the code for this.

The function-name would then be `GetPrimaryConditionsSubset()`.

I don't see a lot of value in getting a list of Derived conditions. The classes that those conditions derive from should already have their own functions to get the relevant data out of those classes.